docs/HowSwiftImportsCModules.md

```diff
@@ -0,0 +1,922 @@
+var ForgotAtHome: HomeworkExcuse { get }
+var ThoughtItWasDueNextWeek: HomeworkExcuse { get }
+
+enum InputDevice : UInt32 {
+  init?(rawValue: UInt32)
+  var rawValue: UInt32 { get }
+  typealias RawValue = UInt32
+  case Keyboard
+  case Mouse
+  case Touchscreen
+}
+
+@_frozen
+enum CardinalDirection : UInt32 {
+  init?(rawValue: UInt32)
+  var rawValue: UInt32 { get }
+  typealias RawValue = UInt32
+  case East
+  case West
+  case North
+  case South
+}
+```
+
+C enums that are not marked as open or closed are mapped to structs since Swift
+1.0. At that time, we realized that mapping C enums to Swift enums is not
+correct if the Swift compiler makes Swift-like assumptions about such imported
+enums. Specifically, Swift compiler assumes that the only allowed bit patterns
+of an enum value are declared in enum's cases. This assumption is valid for
+frozen Swift enums (which were the only flavor of enums in Swift 1.0). However,
+this assumption does not hold for C enums, for which any bit pattern is a valid
+value; this assumption was creating an undefined behavior hazard. To resolve
+this issue in Swift 1.0, C enums were imported as Swift structs by default, and
+their enumerators were exposed as global computed variables. Objective-C enums
+declared with `NS_ENUM` were assumed to have "enum nature" and were imported as
+Swift enums.
+
+The concept of open enums was added in Swift 5 ([SE-0192 Handling Future
+Enum
+Cases](https://github.com/apple/swift-evolution/blob/master/proposals/0192-non-exhaustive-enums.md)),
+but that proposal did not change the importing strategy of non-annotated C
+enums, in part because of source compatibility concerns. It might be still
+possible to change C enums to be imported as open Swift enums, but as the time
+passes, it will be more difficult to change.
+
+Another feature of C enums is that they expose integer values to the user;
+furthermore, enum values are implicitly convertible to integers. Swift enums are
+opaque by default. When imported in Swift, C enums conform to the
+`RawRepresentable` protocol, allowing the user to explicitly convert between
+numeric and typed values.
+
+```swift
+// Converting enum values to integers and back.
+
+var south: CardinalDirection = .South
+// var southAsInteger: UInt32 = south // error: type mismatch
+var southAsInteger: UInt32 = south.rawValue // = 3
+var southAsEnum = CardinalDirection(rawValue: 3) // = South
+```
+
+## Typedefs
+
+C typedefs are generally mapped to Swift typealiases, except for a few common C
+coding patterns that are handled in a special way.
+
+```c
+// C header.
+
+// An ordinary typedef.
+typedef int Money;
+
+// A special case pattern that is mapped to a named struct.
+typedef struct {
+  int x;
+  int y;
+} Point;
+```
+
+```swift
+// C header imported in Swift.
+
+typealias Money = Int
+
+struct Point {
+  var x: Int32 { get set }
+  var y: Int32 { get set }
+  init()
+  init(x: Int32, y: Int32)
+}
+```
+
+# Macros
```

Comment:
See also Apple's documentation about this topic: [Using Imported C Macros in Swift](https://developer.apple.com/documentation/swift/imported_c_and_objective-c_apis/using_imported_c_macros_in_swift)

Reply:
Added.